app/controllers/decks_controller.rb

```diff
@@ -41,18 +41,12 @@ def public
   end
 
   def marketplace
-    @prodecks = Deck.where(deck_type_id: 4).last(15)
-    @pro_decks = Deck.where(deck_type_id: 4).last(10)
-    @top_decks = Deck.where(is_public: true).where('decks.created_at >= ?', 1.week.ago).
-                 group(:unique_deck_id).
-                 joins(:unique_deck).
-                 joins(:user).
-                 where("unique_decks.num_matches >= ?", 30).
-                 sort_by { |deck| deck.unique_deck.winrate || 0 }.
-                 last(20).
-                 reverse.
-                 to_a
-    
+    @prodecks = Deck.where(deck_type_id: 4).last(25)
+    @top_decks = []
+    Rails.cache.read('top_decks').each do |deck_id|
+      @top_decks << Deck.find(deck_id)
```

Comment:
We can just store the `[[deck_name, id(or slug), author], [next deck]].`

This way we don't need to do these queries.